odl/solvers/functional/default_functionals.py

```diff
@@ -611,6 +561,88 @@ def __repr__(self):
         return '{}({!r})'.format(self.__class__.__name__, self.domain)
 
 
+class IndicatorZero(Functional):
+
+    """The indicator function on zero.
+
+    The function has a constant value if the input is zero, otherwise infinity.
+    """
+
+    def __init__(self, space, constant=0):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        space : `LinearSpace`
+            Domain of the functional.
+        constant : element in ``domain.field``, optional
+            The constant value of the functional
+
+        Examples
+        --------
+        >>> space = odl.rn(3)
+        >>> func = IndicatorZero(space)
+        >>> func([0, 0, 0])
+        0
+        >>> func([0, 0, 1])
+        inf
+
+        >>> func = IndicatorZero(space, constant=2)
+        >>> func([0, 0, 0])
+        2
+        """
+        self.__constant = constant
+        super().__init__(space, linear=False)
+
+    @property
+    def constant(self):
+        """The constant value of the functional if ``x=0``."""
+        return self.__constant
+
+    def _call(self, x):
+        """Apply the functional to the given point."""
+
+        if x.norm() == 0:
+            # In this case x is the zero-element.
+            return self.constant
+        else:
+            return np.inf
+
+    @property
+    def convex_conj(self):
+        """The convex conjugate functional.
+
+        Notes
+        -----
+        By the Fenchel-Moreau theorem the convex conjugate functional
+        is the constant functional [BC2011]_.
+        """
+        return ConstantFunctional(self.domain, -self.constant)
+
+    @property
+    def proximal(self):
+        """Return the proximal factory of the functional.
+
+        This is the zero operator.
+        """
+        def zero_proximal(sigma=1.0):
+            """Proximal factory for zero operator.
+
+            Parameters
+            ----------
+            sigma : positive float, optional
+                Step size parameter.
+            """
+            return ZeroOperator(self.domain)
+
+        return zero_proximal
+
+    def __repr__(self):
+        """Return ``repr(self)``."""
+        return '{}({!r}, {!r})'.format(self.__class__.__name__,
+                                       self.domain, self.value)
```

Comment:
Skipping the "print only if needed" step, trying to avoid if statements across the board that makes the code harder to change. This is for debugging anyway.